### PR TITLE
Remove unnecessary member initialization calls

### PR DESCRIPTION
**Pull request by bernhardmgruber**
This should be a non-functional change.

### CODE REVIEW

File: include/alpaka/acc/AccGenericSycl.hpp [hunk 1]
```diff
@@ -81,16 +81,10 @@ namespace alpaka
...
             , BlockSharedMemStGenericSycl{st_shared_acc}
             , BlockSyncGenericSycl<TDim>{work_item}
-            , IntrinsicGenericSycl{}
-            , MemFenceGenericSycl{}
-#    ifdef ALPAKA_DISABLE_VENDOR_RNG
-            , rand::RandDefault{}
-#    else
+#    ifndef ALPAKA_DISABLE_VENDOR_RNG
             , rand::RandGenericSycl<TDim>{work_item}
 #    endif
```
Comment (fwyzard) [inline include/alpaka/acc/AccGenericSycl.hpp:89]: I understand you wanted to keep only the strictly necessary initialisations.
Just for this case where one of the two `#ifdef` branches requires initialisation, would it make sense to keep also the other ?

Comment (bernhardmgruber) [reply include/alpaka/acc/AccGenericSycl.hpp:89]: It does not make sense for me. I would only keep the code that really runs something.

Comment (fwyzard) [reply include/alpaka/acc/AccGenericSycl.hpp:89]: So... I should go ahead and remove all the comments and all the code that is there only to silence warnings ?

Comment (fwyzard) [reply include/alpaka/acc/AccGenericSycl.hpp:89]: Anyway, whatever.

Comment (bernhardmgruber) [reply include/alpaka/acc/AccGenericSycl.hpp:89]: > So... I should go ahead and remove all the code that is there only to silence warnings ?

I also dislike that sort of code. I am saddened that we have to add it to workaround compiler warnings.

But I am always happy if we can agree on dropping older compilers to remove some of that baggage!

Comment (j-stephan) [reply include/alpaka/acc/AccGenericSycl.hpp:89]: IMO this is a question of style / preference. Let us discuss this in tomorrow's meeting.

**Review by SimeonEhrig — APPROVED** [review]

File: include/alpaka/acc/AccGenericSycl.hpp (hunk 1 above)
Comment (bernhardmgruber) [reply include/alpaka/acc/AccGenericSycl.hpp:89]: Discussed during today's alpaka meeting: no objections.

**Pull request merged**